open-hackathon-server/src/hackathon/hack/hackathon_manager.py

```diff
@@ -771,8 +770,12 @@ def __get_hackathon_detail(self, hackathon, user=None):
                 detail["stat"]["like"] = stat.count
 
         if user:
+            hackathon_like = UserHackathon.objects(hackathon=hackathon,user=user).only('like').first()
+            if hackathon_like and hackathon_like.like:
+                detail['like'] = hackathon_like.like
+
             detail["user"] = self.user_manager.user_display_info(user)
-            detail["user"]["is_admin"] = user.is_super or hackathon.creator.id == user.id
+            detail["user"]["is_admin"] = self.admin_manager.is_hackathon_admin(hackathon.id, user.id)
```

Comment:
It seems that `is_hackathon_admin` and `hackathon_like` query the same db collection with same parameter. Can we combine the two query into one?

Please also check the caller of method `__get_hackathon_detail`, if it's called in a loop, it's much better we query db only once outside the loop rather than query the db one by one.